src/main/java/org/webshar/hrms/service/LeaveAllocationService.java

```diff
@@ -101,46 +101,56 @@ public LeaveAllocationResponse assignLeavesToAnEmployee(
 
   public LeaveAllocationResponse updateEmployeesAllocatedLeavesByEmployeeIdAndLeaveType(
       final Long id, final EmployeeLeaveAllocationUpdateRequest employeeLeaveAllocationUpdateRequest)
-      throws EntityNotFoundException, EntityAlreadyExistsException {
+      throws EntityNotFoundException, EntityAlreadyExistsException
+  {
     Employee employee = employeeService
-        .getEmployeeByEmployeeId(employeeLeaveAllocationUpdateRequest.getEmployeeId());
+        .getEmployeeById(employeeLeaveAllocationUpdateRequest.getEmployeeId());
     LeaveType leaveType = leaveTypeService
         .getLeaveTypeById(employeeLeaveAllocationUpdateRequest.getLeaveTypeId());
 
-    isStartAndEndDateAreOverlappingWithExistingRecordForGivenEmployeeAndLeaveTypeId(
-        employee.getId(), leaveType.getId(), employeeLeaveAllocationUpdateRequest.getStartDate(),
-        employeeLeaveAllocationUpdateRequest.getEndDate());
-
-    LeaveAllocation employeeLeaveAllocationToUpdate =
-        leaveAllocationRepository.findById(id)
-            .orElseThrow(
-                () -> new EntityNotFoundException(ErrorMessageConstants.EMPLOYEE_LEAVE_ALLOCATED_BY_ID_NOT_FOUND));
-
-    LeaveAllocation updatedLeaveAllocation =
-        leaveAllocationBuilder.buildFromRequest(employeeLeaveAllocationUpdateRequest,
-            employeeLeaveAllocationToUpdate, employee, leaveType);
+    if (employeeLeaveAllocationUpdateRequest.getStartDate() != null
+        && employeeLeaveAllocationUpdateRequest.getEndDate() != null)
```

Comment:
Why this nonnull check present here. It is expected to always have value here. When do you think this value can be null and what is expected in that scenario.

Reply:
Suppose an employee already have 5 additionalLeaves allocated for a time period 01-01-2020 to 31-12-2020. Now if HR wants to update/add the extra additionalLeaves for the same time period then in the update request the date parameters are optional. 
Assumption No 1) If the dates are coming in the request means the original dates rang wants to update. In that case we are checking overlapping conditions.
Assumption No 2) If dates are not present in the request means just update other parameters like additionalLeaves, carriedLeaves or allottedLeaves.

Reply:
Ok got it. Considering this line 121 findById seems incorrect

Reply:
Agreed, bug fixed.